### DISCUSSION TITLE
Using precompiled packages

### DISCUSSION THREAD

**Discussion by Nesavi** [pyscripter/python4delphi#475] [Q&A] [answered]
Am I correct in assuming that precompiled packages can't be used when generating pyd-files? I've tried, and the script importing from my pyd-file fails.

**Comment by pyscripter** [comment]
Not sure what you are trying to do.  Could you give a detailed minimal example?

**Reply by Nesavi** [reply]
Yes, I'd be happy to. Do you have a shared drive that I can upload a sample to?

**Reply by pyscripter** [reply]
You can upload files here.  A zipped project would do.

**Reply by Nesavi** [reply]
Thanks for taking time to answer my question. It's much appreciated :)

The attached zip-file contains to project. The only difference between them, apart from their names, is that the project 'DelphiFromPython_DCP.dproj' has 'Link with runtime packages' set to true:
![image](https://github.com/pyscripter/python4delphi/assets/39985908/9fdbe612-9515-4c12-a9c1-9a1b7e305135)

[dcp_test.zip](https://github.com/user-attachments/files/15805161/dcp_test.zip)

Python 3.12.3 was used:

![image](https://github.com/pyscripter/python4delphi/assets/39985908/153b166d-0c78-481b-a859-aeaa5bab1b08)

**Reply by pyscripter** [reply, marked as answer]
Firstly, you should not be using LoadDLL in your extension modules but LoadDllInExtensionModule.  See https://github.com/pyscripter/python4delphi/blob/master/Modules/DemoModule/uMain.pas as a guide.

Regarding runtime packages, is the directory containing them in the path?
Even if it is there was a change in python when searching for dlls.  The Windows path is no longer searched.
See 
https://docs.python.org/3/library/os.html#os.add_dll_directory
and https://stackoverflow.com/questions/67805339/is-the-function-os-add-dll-directory-adding-directories-permanently.

You can try:
- copying all runtime packages in the same directory as the pyd file.
- or use the os.add_dll_directory function before importing the pyd extension module.
